src/DataDepsGenerators.jl

```diff
@@ -125,14 +167,18 @@ function format_meta(data::Any, label="")
     return "\n" * indent(label * string(data))
 end
 
+
 function generate(repo::DataRepo,
                   dataname,
                   shortname = nothing
     )
-
-    meta = find_metadata(repo, dataname, shortname)
-
-    body(meta)
+    generators = []
+    generator_meta = []
+    push!(generators, repo)
+    for ii in generators
+        push!(generator_meta, find_metadata(ii, dataname, shortname))
+    end
+    aggregate(generator_meta)
```

Comment:
Stick into  `utils.jl`
```
"""
    leaf_subtypes(T)
Returns all the nonabstract types decedent from `T`.
"""
function leaf_subtypes(T)
       if isleaftype(T)
           T
       else
           vcat(leaf_subtypes.(subtypes(T))...)
       end
end
```  

Into the main file for generation stick something like

```
function generate(repo::DataRepo, dataname; kwargs...)
    generate([repo], dataname; kwargs...)
end

function generate(dataname; kwargs...)
    all_repos = [T() for T in leaf_subtypes(DataRepo)]
    generate(all_repos , dataname; kwargs...)
end


function generate(repos, dataname, shortname = nothing, show_failures=false)
	retrieved_metadatas = MetaData[]
	failures = Tuple{DataRepo, Exception}[]
	
	# Get all the metadata we can
	@sync for repo in repos
		@async try
			metadata = find_metadata(repo, dataname, shortname)
			push!(retrieved_metadatas, metadata)
		catch err
			push!(failures, (repo, err))
		end
	end
	
	# Displace errors if required
	if length(retrieved_metadatas) == 0 || show_failures
		for (repo, err) in failures
			println(repo, " failed due to")
			println(err)
			println()
		end
	end
	
	# merge them
	aggregate(retrieved_metadatas)
end
```

I haven't tested this code.
But hopefully you can get the idea.